sound/soc/sof/sof-priv.h

```diff
@@ -249,6 +249,9 @@ struct snd_sof_dsp_ops {
 	void (*set_mach_params)(const struct snd_soc_acpi_mach *mach,
 				struct device *dev); /* optional */
 
+	/* client ops */
+	void (*client_register)(struct snd_sof_dev *sdev); /* optional */
```

Comment:
is a special operation really needed for this? Wouldn't it be possible for machines to just register their clients in their `.probe()`?

Reply:
@lyakh the machine drivers possibly would not have information to register all the clients. Like I mentioned in the commit message, some clients are based on ACPI info and some others  based on what the DSP advertises in terms of capabilities.

Reply:
And oh the other thing to mention is that the machine driver itself would be registered within the client (an audio clien will be added and the current code moved out of the core)